cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,311 @@
+			if err != nil {
+				return nil, fmt.Errorf("failed to to extract raw cloudtralEvent details: %w", err)
+			}
+			userArn := raw.UserIdentity.SessionContext.SessionIssuer.Arn
+			regexOdj := regexp.MustCompile(mergedRegex)
+			matchesUsername := false
+			matchesArn := false
+
+			if shouldFilter {
+				if event.Username != nil {
+					matchesUsername = regexOdj.MatchString(*event.Username)
+				}
+				if userArn != "" {
+					matchesArn = regexOdj.MatchString(userArn)
+
+				}
+
+				if matchesUsername || matchesArn {
+					continue
+					// skips entry
+				}
+			}
+
+			filteredEvents = append(filteredEvents, event)
+		}
+	}
+
+	return &filteredEvents, nil
+}
+
+// PrintEvents prints the details of each event in the provided slice of events.
+// It takes a slice of types.Event
+func printEvents(filteredEvent []types.Event, printUrl bool, raw bool) {
+	for _, event := range filteredEvent {
+
+		if raw {
+			if event.CloudTrailEvent != nil {
+				fmt.Printf("\n%v	\n", *event.CloudTrailEvent)
+			}
+		} else {
+			if event.EventName != nil {
+				fmt.Printf("%v |", *event.EventName)
+			} else {
+				fmt.Println("<not available> |")
+			}
+
+			if event.EventTime != nil {
+				fmt.Printf("%v |", event.EventTime.String())
+			} else {
+				fmt.Println("<not available> |")
+			}
+
+			if event.Username != nil {
+				fmt.Printf("User: %v |\n", *event.Username)
+			} else {
+				fmt.Println("User: <not available> |")
+			}
+
+			if printUrl && event.CloudTrailEvent != nil {
+				details, err := extractUserDetails(event.CloudTrailEvent)
+				if err == nil {
+					fmt.Printf("EventLink: %v\n\n", generateLink(*details))
+				} else {
+					fmt.Println("EventLink: <not available>")
+				}
+			}
+
+		}
+
+	}
+
+}
+
+func (o *LookupEventsOptions) run() error {
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return fmt.Errorf("unable to create connection to ocm: %w", err)
+	}
+	defer connection.Close()
+
+	cluster, err := utils.GetClusterAnyStatus(connection, o.clusterID)
+	if err != nil {
+		return err
+	}
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return errors.New("this command is only available for AWS clusters")
+	}
+
+	cfg, err := osdCloud.CreateAWSV2Config(connection, cluster)
+	if err != nil {
+		return err
+	}
+
+	cloudtrailClient := cloudtrail.NewFromConfig(cfg)
+	arn, accountId, err := whoami(*sts.NewFromConfig(cfg))
+	if err != nil {
+		return err
+	}
+
+	starttime, err := parseDurationToUTC(o.startTime)
+	if err != nil {
+		return err
+	}
+
+	lookupOutput, err := getWriteEvents(starttime, cloudtrailClient)
+	if err != nil {
+		return err
+	}
+	Ignore, err := config.LoadCloudTrailConfig()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Checking write event history since %v for AWS Account %v as %v \n", starttime, accountId, arn)
+	fmt.Printf("\n[+] Fetching %v Event History...\n", cfg.Region)
```

Comment:
**Suggestion:** move this above the querying logic, so the user knows this is happening. Currently, this is printed after getting the events.